src/features/cardPreview/CardPreview.tsx

```diff
@@ -0,0 +1,90 @@
+import {
+  StyledContainer,
+  StyledIconWrap,
+  StyledMagnetic,
+  StyledLogoWrap,
+  StyledCardNumberWrap,
+  StyledCardNumber,
+  StyledExpirationPeriod,
+} from "./CardPreview.css";
+import {
+  CardNumberPosition,
+  ExpirationPeriod,
+} from "../../\btypes/index.types";
+import { useState } from "react";
+import { useEffect } from "react";
+import { INITIALIZE_VALUE } from "../../shared/constants/constant";
+
+type CardNumber = Record<CardNumberPosition, string>;
+
+type CardPreviewProps = {
+  cardNumber: CardNumber;
+  expirationPeriod: Record<ExpirationPeriod, string>;
+};
+
+const CARD_IDENTIFYING_NUMBER = {
+  VISA: 4,
+  MASTERCARD: {
+    MIN: 51,
+    MAX: 55,
+  },
+};
+
+function getIdentifyFns(id: string) {
+  return [
+    {
+      identify: () => Number(id[0]) === CARD_IDENTIFYING_NUMBER.VISA,
+      logoSrc: "./images/Visa.svg",
+    },
+    {
+      identify: () =>
+        Number(id) >= CARD_IDENTIFYING_NUMBER.MASTERCARD.MIN &&
+        Number(id) <= CARD_IDENTIFYING_NUMBER.MASTERCARD.MAX,
+      logoSrc: "./images/Mastercard.svg",
+    },
+  ];
+}
+
+function CardPreview({ cardNumber, expirationPeriod }: CardPreviewProps) {
+  const [logoSrc, setLogoSrc] = useState(INITIALIZE_VALUE);
+
+  useEffect(() => {
+    const id = cardNumber.first.slice(0, 2);
+
+    const identifyFns = getIdentifyFns(id);
+    const identifiedLogoSrc = identifyFns.find((fn) => fn.identify());
+    setLogoSrc(identifiedLogoSrc?.logoSrc || "");
+  }, [cardNumber]);
+
+  return (
+    <StyledContainer>
+      <StyledIconWrap>
+        <StyledMagnetic></StyledMagnetic>
+        {logoSrc !== INITIALIZE_VALUE ? (
+          <StyledLogoWrap>
+            <img
+              src={logoSrc}
+              alt="logo"
+              style={{ width: "100%", height: "100%" }}
+            />
+          </StyledLogoWrap>
+        ) : null}
```

Comment:
```suggestion
        {logoSrc && (
          <StyledLogoWrap>
            <img
              src={logoSrc}
              alt="logo"
              style={{ width: "100%", height: "100%" }}
            />
          </StyledLogoWrap>
        )}
```

logoSrc가 빈 문자열이므로 && 형식으로 해보는 것은 어떤가요?? 
우디의 현재 코드는 좀 더 명시적으로 조건을 보여주려고 하신 것일까요? 🤔

Reply:
좋은데요~?